lib/Dialect/Seq/Transforms/LowerSeqToSV.cpp

```diff
@@ -282,29 +283,34 @@ void FirRegLower::lower() {
   if (regs.empty())
     return;
 
-  // Lower the regs to SV regs.
-  SmallVector<RegLowerInfo> toInit;
-  for (auto reg : llvm::make_early_inc_range(regs))
-    toInit.push_back(lower(reg));
+  // Lower the regs to SV regs. Group them by initializer and reset kind.
+  SmallVector<RegLowerInfo> randomInit, presetInit;
+  llvm::MapVector<Value, SmallVector<RegLowerInfo>> asyncResets;
+  for (auto reg : llvm::make_early_inc_range(regs)) {
+    auto svReg = lower(reg);
+    if (svReg.preset)
+      presetInit.push_back(svReg);
+    else if (!disableRegRandomization)
+      randomInit.push_back(svReg);
```

Comment:
Alternatively, why not just emit both? The random is `ifdef` guarded and the preset follows the randomization and is not `ifdef` guarded.

This logic for emission should already work for this below.